### PR TITLE
Extension exposer replacement and extensions.js changes

### PR DESCRIPTION
**Pull request by yuri-kiss**
Made some changes to extensions.js to fix my github username and fix some notes, added warning to scope variables for obvious reason, and the main reason for this pr, I(please read the comments for further context, it gives more than enough to understand whats going on, comments marked as abuse were by me; as I thought I should not have said it, and hence "abuse" because its rude).

### CODE REVIEW

**Comment by TheShovel** [comment]
the replacement for extension exposer does not need to exist. not only does i have missing blocks but it also misses the example functions, and the blocks are worded worse. please remove that from this PR or im not merging it

**Comment by TheShovel** [comment]
im not the say all be all but this pr does concern me. "its not worded like scratch" doesnt mean its not worded well, and the extension is not poorly written. please lets stop arguing over a text file, because im not going to change my mind

**Comment by TheShovel** [comment]
> > im not the say all be all but this pr does concern me. "its not worded like scratch" doesnt mean its not worded well, and the extension is not poorly written. please lets stop arguing over a text file, because im not going to change my mind
> 
> im not arguing over a text file; I am responding to your complaints and feedback, and if you are not open to suggestion or change then why are you contributing to a open source repo in the first place? another thing is that it is poorly written like wise the way its formatted and lack there of checking for anything before hand, you should not rely all on pm's error catching.

if pms error catching is enough then why bother with checking again...? its slower. pm already checks for errors. why should i do it again?

**Comment by TheShovel** [comment]
> > > > im not the say all be all but this pr does concern me. "its not worded like scratch" doesnt mean its not worded well, and the extension is not poorly written. please lets stop arguing over a text file, because im not going to change my mind
> > > 
> > > 
> > > im not arguing over a text file; I am responding to your complaints and feedback, and if you are not open to suggestion or change then why are you contributing to a open source repo in the first place? another thing is that it is poorly written like wise the way its formatted and lack there of checking for anything before hand, you should not rely all on pm's error catching.
> > 
> > 
> > if pms error catching is enough then why bother with checking again...? its slower. pm already checks for errors. why should i do it again?
> 
> its not checking for errors on the function; its checking if the function you are try to call exists. error handling was brought up because it throws an error if the function does not exist.

yes, the pm error catching will tell you the function doesnt exist, and for anyone using an extension as technical as this, the pm error is enough

**Comment by TheShovel** [comment]
> > > > > > im not the say all be all but this pr does concern me. "its not worded like scratch" doesnt mean its not worded well, and the extension is not poorly written. please lets stop arguing over a text file, because im not going to change my mind
> > > > > 
> > > > > 
> > > > > im not arguing over a text file; I am responding to your complaints and feedback, and if you are not open to suggestion or change then why are you contributing to a open source repo in the first place? another thing is that it is poorly written like wise the way its formatted and lack there of checking for anything before hand, you should not rely all on pm's error catching.
> > > > 
> > > > 
> > > > if pms error catching is enough then why bother with checking again...? its slower. pm already checks for errors. why should i do it again?
> > > 
> > > 
> > > its not checking for errors on the function; its checking if the function you are try to call exists. error handling was brought up because it throws an error if the function does not exist.
> > 
> > 
> > yes, the pm error catching will tell you the function doesnt exist, and for anyone using an extension as technical as this, the pm error is enough
> 
> smh, I cant see why your contributing to open source if you cant take any form of feedback or updates, your just as I thought.

if i would have not contributed, that extension would not exist in the first place

**Comment by JeremyGamer13** [comment]
you cant get errors so you cant use the try catch block
original:
image
new:
image

**Comment by TheShovel** [comment]
if you wanted to update it you would have not done this, and you would have updated the current extension instead. think about how i feel when someone just randomly comes in and says "your code bad mine better im going to replace you". when i made this extension i made it this way with a reason
- yours is a lot slower in large loops
- having boolean blocks separate to reporter blocks helps with reading large threads of code
- error checking in this case removes from the ease of understanding why something did not work
-  the way you worded yours is done just because you wanted to word it differently. literally has no benefit

the way you responded to me comes along as "i dont like you" especially after saying i am "just as you thought"

updating the existing extension would have been a better idea if you wanted to improve it in any way, even though as you said this is a very simple extension, which is best to have a simple implementation for

image

File: static/extensions/TheShovel/extexp.js [hunk 1]
```diff
@@ -1,66 +1,86 @@
...
-        return args.INPUT
+      return Cast.toString(args.INPUT || ''));
+    }
+    run({ FUNCNAME, EXTLIST, INPUT }, util, blockJSON) {
+      XTLIST = Cast.toString(EXTLIST);
+      FUNCNAME = Cast.toString(FUNCNAME);
+      // If the function does not exist then it is not referenced as a real block, or the extension is not global (fallback)
+      return (runtime._primitives[`${EXTLIST}_${FUNCNAME}`] || runtime[`ext_${EXTLIST}`][FUNCNAME])(this._parseJSON(Cast.toString(INPUT)), util, blockJSON);
     }
+    runcommand() {}
```
Comment (TheShovel) [inline static/extensions/TheShovel/extexp.js:81]: make sure that these redirect to the new function so projects don't break

Comment (TheShovel) [reply static/extensions/TheShovel/extexp.js:82]: did not know that func was a property you could set

**Pull request merged**